tsdb/index/index.go

```diff
@@ -1505,6 +1510,26 @@ func (r *Reader) LabelValues(name string) ([]string, error) {
 	return values, nil
 }
 
+// LabelValueFor returns value of given label value of series referred to by ID.
+func (r *Reader) LabelValueFor(id uint64, label string) (string, error) {
+	offset := id
+	// In version 2 series IDs are no longer exact references but series are 16-byte padded
+	// and the ID is the multiple of 16 of the actual position.
+	if r.version == FormatV2 {
+		offset = id * 16
+	}
+	d := encoding.NewDecbufUvarintAt(r.b, int(offset), castagnoliTable)
+	if d.Err() != nil {
```

Comment:
Maybe do the `d.Get()` before this and then check the `d.Err()` error. We should do after `d.Get()` anyway.

Reply:
is this what you had in mind? d2191c5a5e67d373e0f972a173e6fcb5cfcab578

Reply:
Yes